lib/autoupdate.ps1

```diff
@@ -431,3 +424,101 @@ function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $
         Write-Host -f DarkGray "No updates for $app"
     }
 }
+
+## Helper Functions
+
+function PropertyHelper {
+    <#
+    .SYNOPSIS
+        Helper of updating property
+    .DESCRIPTION
+        Update manifest property (String, Array or PSCustomObject).
+    .PARAMETER Property
+        Property to be updated
+    .PARAMETER Value
+        New property values
+        Update line by line
+    .OUTPUTS
+        System.Object[]
+            The first element is new property, the second element is change flag
+    #>
+    param (
+        [Object]$Property,
+        [Object]$Value
+    )
+    $hasChanged = $false
+    if ($Property -is [String]) {
+        $Value = $Value -as [String]
+        if (($null -eq $Value) -or ($value -eq $Property)) {
+            return $Property, $false
+        } else {
+            return $Value, $true
+        }
+    } elseif ($Property -is [Array]) {
+        $Value = @($Value)
+        for ($i = 0; $i -lt [Math]::Min($Property.Length, $Value.Length); $i++) {
+            $Property[$i], $hasItemChanged = PropertyHelper -Property $Property[$i] -Value $Value[$i]
+            $hasChanged = $hasChanged -or $hasItemChanged
```

Comment:
Should not there be some `break`?

Reply:
Because then you could potentionally skip some properties. Right?

You are returning only single property and boolean in #L463.

Reply:
This is for one property, and `$Property[$i]` is one item (or line) in this property. This `for` loop replace first `$Value.Length` lines or items of the property, or all lines or items if `$Property.Length` is less then or equal to `$Value.Length`.

Reply:
Then you could easily break as soon as one item in property was changed.  To skip remaining.

Reply:
But we must go through all items to ensure they are updated, at least `$Value.Length`, at most `$Property.Length`. We cannot skip this check and replace.

Reply:
Ohh... good. Sorry I missed that recursive call.